app/scripts/controllers/transactions/lib/util.js

```diff
@@ -173,6 +178,11 @@ export function validateTxParams(txParams, eip1559Compatibility = true) {
           'gasPrice',
         );
         ensureFieldIsString(txParams, 'maxFeePerGas');
+        if (!value.match(/^0x[a-fA-F0-9]+$/u)) {
```

Comment:
Since this code fragment seems to be duplicated few times, can we extract this into a simple function (i.e. ensureValueIsValidHexString())? For error messages try passing the param for the fieldName, etc.